### ISSUE TITLE
Cannot delete input files after using gdal_utils on Windows

### ISSUE THREAD

**Issue by ranghetti** [r-spatial/sf#1452]
Hi,
I encountered this issue after a problem experienced by @jamesfitton in https://github.com/ranghetti/sen2r/issues/336: on Windows, after launching `gdal_utils()` on a source raster, this file seems to be kept in use by the R session and so cannot be deleted until the current session is stopped. This error does not occur using a GDAL system call, nor using `gdal_utils()` on Linux.

Here a reproducible example:
```r
# Check that input file can be commonly deleted
in_path <- tempfile(fileext = ".tif")
file.copy(system.file("tif/L7_ETMs.tif", package = "stars"), in_path)
file.exists(in_path) # TRUE
file.remove(in_path) # FALSE
file.exists(in_path) # TRUE

# Issue after using gdal_utils
file.copy(system.file("tif/L7_ETMs.tif", package = "stars"), in_path)
out_path <- tempfile(fileext = ".tif")
sf::gdal_utils("translate", in_path, out_path, options = c("-of", "GTiff"))
file.exists(in_path) # TRUE
file.remove(in_path) # FALSE (see warning)
file.exists(in_path) # TRUE

# Issue occurs only in the current R session
.rs.restartR()
file.remove(in_path) # TRUE
file.exists(in_path) # FALSE

# Issue does not occur calling a runtime GDAL
(gdaltranslate_path <- Sys.which("gdal_translate")) # Check this worked properly
file.copy(system.file("tif/L7_ETMs.tif", package = "stars"), in_path)
system(paste(gdaltranslate_path, in_path, out_path, "-of GTiff"))
file.exists(in_path) # TRUE
file.remove(in_path) # TRUE
file.exists(in_path) # FALSE
```
This is the warning being returned:
```
Warning message:
In file.remove(in_path) :
  cannot remove file 'C:\Users\LRANGH~1\AppData\Local\Temp\2\Rtmp0CCXxE\file2bd44b80b0.tif', reason 'Permission denied'
```

Here the session info:
```
> devtools::session_info()
- Session info ----------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.5.3 (2019-03-11)
 os       Windows Server >= 2012 x64  
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  Italian_Italy.1252          
 ctype    Italian_Italy.1252          
 tz       Europe/Berlin               
 date     2020-07-17                  

- Packages --------------------------------------------------------------------------------------------------
 package     * version date       lib source        
 sf            0.9-5   2020-07-14 [1] CRAN (R 3.5.3)
 [...]

[1] C:/Users/Public/Documents/R/win-library/3.5
[2] C:/Program Files/R/R-3.5.3/library
```
@jamesfitton reported the original issue using R (4.0.1) in Jupyter Notebook on Windows and a Anaconda Environment.

Thanks,

**Comment by edzer** [comment]
Yes, this was done intentionally to fix #1352:
https://github.com/r-spatial/sf/blob/02919c3e47e4c0edc0893116a757c7dfb662e7a4/src/gdal_utils.cpp#L113-L116

I'd be interested in a fix that solves that one, and this one!

**Comment by edzer** [comment]
Turns out it is the order of closing resources that counted...
